cluster/src/main/java/com/hartwig/pipeline/sbpapi/SbpRestApi.java

```diff
@@ -113,10 +115,19 @@ public AddFileApiResponse postFile(final SbpFileMetadata metaData) {
         }
     }
 
-    public void linkFileToSample(final int id, final int sampleId) {
-        Map<String, Integer> request = new HashMap<>();
-        request.put("sample_id", sampleId);
-        submitJson(POST, api().path(format("%s/%d/sample", FILES, id)), request, Response.Status.CREATED);
+    public void linkFileToSample(final int id, final String barcode) {
+        try {
+            SbpSample sample = ObjectMappers.get().<List<SbpSample>>readValue(returnOrThrow(api().path(SAMPLES)
+                    .queryParam("barcode", barcode)
+                    .request()
+                    .get()), new TypeReference<List<SbpSample>>() {
+            }).stream().findFirst().orElseThrow();
+            Map<String, Integer> request = new HashMap<>();
+            request.put("sample_id", sample.id());
+            returnOrThrow(submitJson(POST, api().path(FILES).path(String.valueOf(id)).path("sample"), request, Response.Status.CREATED));
```

Comment:
This is an opportunity to loosen the allowable response codes so we can make the API return a more standard response.